core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -159,6 +181,13 @@ String getPreferredIp(List<String> preferredTypes) {
    */
   void forceRefresh() {
     synchronized (instanceDataGuard) {
+      // Don't force a refresh until the current forceRefresh operation
```

Comment:
Would you mind sending a PR with all the logging added here? https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/tree/enocom-debug It was quite useful for debugging and would be good to have in the future.